runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java

```diff
@@ -142,9 +144,21 @@ public long append(@Nonnull Set<UUID> streamIDs, @Nonnull Object object,
                 // If we're here, we succeeded, return the acquired token
                 return tokenResponse.getTokenValue();
             } catch (OverwriteException oe) {
+
+                if (retry-- == 0) {
+                    ILogData ld = new LogData(DataType.DATA, object);
+                    ILogData.SerializationHandle sh = ld.getSerializedForm();
+                    log.error("append[{}]: Overwrite , streams {}, write size {}",
+                            tokenResponse.getTokenValue(),
+                            streamIDs.stream().map(Utils::toReadableId).collect(Collectors.toSet()),
+                            sh.getSerialized().getSizeEstimate());
+                    sh.close();
+                    throw oe;
```

Comment:
Is overwrite exception really the right exception to throw here? How will the client catch it? Maybe you should throw a wrapped TransactionAbortedException like the rest of the error conditions?